### PR TITLE
test: Add Linux smoke tests

### PR DESCRIPTION
**Pull request by nrcventura**
Adds a new test solution and github action that allows us to run smoke tests of the agent in containerized apps so that we can test that the agent is working as expected across a wider variety of systems.

The initial set of tests validates that the agent is working on
*  ubuntu x64 and arm64
* debian x64 and arm64
*  alpine x64 (only)
* centos stream 9 x64 and arm64
* amazon linux x64 and arm64

### CODE REVIEW

File: build/build.ps1 [hunk 1]
```diff
@@ -46,7 +46,7 @@ $solutions = [Ordered]@{
     "$rootDirectory\src\Agent\MsiInstaller\MsiInstaller.sln"                    = @("Configuration=$Configuration;Platform=x86;AllowUnsafeBlocks=true","Configuration=$Configuration;Platform=x64;AllowUnsafeBlocks=true");
     "$rootDirectory\tests\Agent\IntegrationTests\IntegrationTests.sln"          = @("Configuration=$Configuration;DeployOnBuild=true;PublishProfile=LocalDeploy");
     "$rootDirectory\tests\Agent\IntegrationTests\UnboundedIntegrationTests.sln" = @("Configuration=$Configuration;DeployOnBuild=true;PublishProfile=LocalDeploy");
-    "$rootDirectory\tests\Agent\PlatformTests\PlatformTests.sln"                = @("Configuration=$Configuration");
+#    "$rootDirectory\tests\Agent\PlatformTests\PlatformTests.sln"                = @("Configuration=$Configuration");
```
Comment (jaffinito) [inline build/build.ps1:49]: Should we just remove the line?

File: tests/Agent/IntegrationTests/ContainerApplications/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+version: "3"
+
+# The following must be set either in environment variables or via a .env file in the same folder as this file:
```
Comment (jaffinito) [inline tests/Agent/IntegrationTests/ContainerApplications/docker-compose.yml:3]: Love the documentation here!

File: .github/workflows/run_linux_smoke_tests.yml [hunk 3]
```diff
@@ -26,8 +26,39 @@ jobs:
...
+      test_results_path: tests\TestResults
+      integration_tests_shared_project: ${{ github.workspace }}/tests/Agent/IntegrationTests/Shared
+      NR_DOTNET_TEST_SAVE_WORKING_DIRECTORY: 1
+      # Make this variable true to enable extra data-gathering and logging to help troubleshoot test failures, at the cost of additional time and resources
+      enhanced_logging: false
+  
+  
     steps:
+      - name: Docker Setup QEMU
+        uses: docker/setup-qemu-action@v2.2.0
```
Comment (chynesNR) [inline .github/workflows/run_linux_smoke_tests.yml:39]: Do we need to pin this to a specific commit?

File: src/Agent/NewRelic.Api.Agent/NewRelic.Api.Agent.csproj [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net462;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net462;netstandard2.0;net7.0</TargetFrameworks>
```
Comment (chynesNR) [inline src/Agent/NewRelic.Api.Agent/NewRelic.Api.Agent.csproj:3]: Did we determine if this is necessary?

File: tests/Agent/IntegrationTests/ContainerApplications/SmokeTestApp/Dockerfile.amazon [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+RUN dotnet build "SmokeTestApp.csproj" -c Release -o /app/build --os linux -a ${TARGET_ARCH}
+
+FROM build AS publish
+RUN dotnet publish "SmokeTestApp.csproj" -c Release -o /app/publish /p:UseAppHost=false --os linux -a ${TARGET_ARCH}
+
+FROM base AS final
+
+# install asp.netcore - use the Centos 7 package feed because amazon is rhel based
+RUN rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm
+RUN yum install aspnetcore-runtime-7.0 -y
```
Comment (chynesNR) [inline tests/Agent/IntegrationTests/ContainerApplications/SmokeTestApp/Dockerfile.amazon:24]: Would it be worth making the runtime a parameter, in order to make transitioning to .NET 8 (etc) easier later?

File: .github/workflows/run_linux_smoke_tests.yml (hunk 3 above)
Comment (tippmar-nr) [reply .github/workflows/run_linux_smoke_tests.yml:39]: Good catch, yeah we'll need to pin it or the security scanner will get mad...

File: src/Agent/NewRelic.Api.Agent/NewRelic.Api.Agent.csproj (hunk 4 above)
Comment (nrcventura) [reply src/Agent/NewRelic.Api.Agent/NewRelic.Api.Agent.csproj]: It's not necessary. The API reference is used for some shared WCF service code, so I reverted that change and updated the shared project to conditionally reference the API project.

**Review by chynesNR — APPROVED** [review]
Very nice!

**Review by jaffinito — APPROVED** [review]

**Pull request merged**